components/CommentForm.jsx

```diff
@@ -0,0 +1,46 @@
+import { useState } from "react"
+import { postComment } from "../api"
+
+const CommentForm = ({articleId, setComments}) => {
+const [formData, setFormData] = useState({username:"", body:""})
+const [isLoading, setIsLoading] = useState(false)
+const [success, setSuccess] = useState(false)
+    
+const handleChange = (e) => {
+setFormData({...formData, [e.target.name === "name" ? "username" : e.target.name]: e.target.value})
+}
+
+const handleSubmit = (e) => {
+    e.preventDefault()
+    setIsLoading(true)
+    postComment(articleId, formData).then((newComment)=> {
+        handleNewComment(newComment)
+        setSuccess(true)
+        setIsLoading(false)
+        setFormData({ username: "", body: "" });
+    }).catch((error) => {
+        console.error('Error submitting comment:', error);
+        setIsLoading(false);
+      });
+}
+const handleNewComment = (newItem) => {
```

Comment:
A bit unnecessary perhaps.  Could have a simple setState functionality that is in this function moved to line 17.